### PR TITLE
Document SSL_set_session(pSSL, NULL) behaviour

### PR DESCRIPTION
**Pull request by EasySec**
Found a mistake in my code because I was expecting that SSL_set_session() on an uninitialized SSL_SESSION* would return 0.
What is your opinion ?
Should the ssl internal session be released in that case ?
Not sure about what I read from the man page.

### CODE REVIEW

**Comment by t8m** [comment]
This is wrong change. As you can see from the failing CI setting a NULL session is not an error.

**Comment by EasySec** [comment]
> This is wrong change. As you can see from the failing CI setting a NULL session is not an error.

Yes, sorry. 
Wouldn't it be worth saying something about this in the man page ?

**Comment by t8m** [comment]
Feel free to submit a documentation PR if you think it should be documented. Basically the call with NULL is to clear the existing session and not replace it with any new one.

File: doc/man3/SSL_set_session.pod [hunk 1]
```diff
@@ -21,7 +21,8 @@ with the L<SSL_session_reused(3)> call.
 
 If there is already a session set inside B<ssl> (because it was set with
 SSL_set_session() before or because the same B<ssl> was already used for
-a connection), SSL_SESSION_free() will be called for that session. If that old
+a connection), SSL_SESSION_free() will be called for that session. 
+This is also the case when B<session> is a null pointer. If that old
```
Comment (t8m) [inline doc/man3/SSL_set_session.pod:25]: Please use NULL in uppercase.

Comment (EasySec) [reply doc/man3/SSL_set_session.pod]: Yes, sure, done, thanks.

**Review by kaduk — APPROVED** [review]
I probably would have added a sentence about how NULL `B<session>` can be used to dis-associate a session from an SSL (i.e., why one would exercise the NULL case), but this is accurate and a clear improvement over the previous state, so I'm happy to approve it.

**Comment by EasySec** [comment]
> I probably would have added a sentence ...

That's because your english is perfect. Far from being my case !
;-)

**Comment by t8m** [comment]
Merged to master and 3.0 branches after trimming off EOL whitespace and slightly rewording the commit message. Thank you for your contribution.

**Pull request closed without merging**